src/main.rs

```diff
@@ -26,13 +26,15 @@ fn main() {
         let stdin = io::stdin();
         log::info!("Welcome to Egglog!");
         let mut egraph = EGraph::default();
-        for line in stdin.lock().lines() {
-            let line = line.unwrap_or_else(|_| panic!("Failed to read line from stdout"));
-            match egraph.parse_and_run_program(&line) {
-                Ok(_msgs) => {}
-                Err(err) => {
-                    log::error!("{}", err);
-                }
+        let mut program = String::new();
+        stdin
+            .lock()
+            .read_to_string(&mut program)
+            .unwrap_or_else(|_| panic!("Failed to read program from stdin"));
+        match egraph.parse_and_run_program(&program, true, false) {
```

Comment:
Big change- for convenience, my proofs code assumes it has the entire program. I think this can be changed fixed, but should it? The interactive mode has been replaced by the stdin accepting an entire program in parenthesis.

Reply:
I fixed the REPL!